tests/testthat/test-folder-project-new.R

```diff
@@ -0,0 +1,207 @@
+        '"https://app.crunch.io/api/datasets/3/",',
+        '"https://app.crunch.io/api/datasets/1streaming/",',
+        '"https://app.crunch.io/api/datasets/1/"]}'
+    )
+    test_that("mv dataset", {
+        expect_PATCH(
+            projects() %>%
+                cd("Project One") %>%
+                mv("test ds", "Project Two"),
+            "https://app.crunch.io/api/projects/project2/",
+            move_testds
+        )
+        expect_PATCH(
+            projects() %>%
+                mv(ds, "Project One/Project Two"),
+            "https://app.crunch.io/api/projects/project2/",
+            move_testds
+        )
+        expect_no_request(
+            # That's already where it is
+            projects() %>%
+                mv(ds, "Project One")
+        )
+    })
+    test_that("Legacy methods: datasets<-", {
+        expect_PATCH(
+            datasets(proj[["Project Two"]]) <- ds,
+            "https://app.crunch.io/api/projects/project2/",
+            move_testds
+        )
+    })
+
+    test_that("mv several datasets (with magic select fn)", {
+        expect_PATCH(
+            projects() %>%
+                cd("Project One") %>%
+                mv(contains("a"), "Project Two"),
+            "https://app.crunch.io/api/projects/project2/",
+            '{"element":"shoji:catalog",',
+            '"index":{"https://app.crunch.io/api/datasets/2/":{},',
+            '"https://app.crunch.io/api/datasets/streaming-no-msg/":{}},"graph":[',
+            '"https://app.crunch.io/api/datasets/3/",',
+            '"https://app.crunch.io/api/datasets/1streaming/",',
+            '"https://app.crunch.io/api/datasets/2/",',
+            '"https://app.crunch.io/api/datasets/streaming-no-msg/"]}'
+        )
+    })
+
+    test_that("mv dataset creates a project if necessary", {
+        expect_POST(
+            projects() %>%
+                cd("Project One") %>%
+                mv("test ds", "Group 2"),
+            "https://app.crunch.io/api/projects/project1/",
+            create_group_2
+        )
+    })
+
+    test_that("mv project", {
+        expect_PATCH(
+            projects() %>%
+                cd("Project One") %>%
+                mv("Project Two", "Project Five"),
+            "https://app.crunch.io/api/projects/project5/",
+            '{"element":"shoji:catalog",',
+            '"index":{"https://app.crunch.io/api/projects/project2/":{}},',
+            '"graph":["https://app.crunch.io/api/projects/project2/"]}'
+
+        )
+    })
+
+    test_that("can't move anything to root", {
+        skip("TODO")
+    })
+
+    test_that("mv error handling", {
+        expect_error(
+            proj %>%
+                mv("NOT A GROUP", "Project Two"),
+            "Undefined elements selected: NOT A GROUP"
+        )
+    })
+
+    test_that("rmdir an empty project", {
+        expect_error(
+            proj %>% rmdir("Project Five"),
+            "Must confirm deleting folder" # TODO: should error say "project"?
```

Comment:
I don't think that's necessary because the confirmation prompt uses the folder name: https://github.com/Crunch-io/rcrunch/blob/master/R/shoji-folder.R#L105